node/src/main.rs

```diff
@@ -481,45 +501,64 @@ fn async_main() -> impl Future<Item = (), Error = ()> + Send + 'static {
         // database.
         assert!(*ANCESTOR_COUNT >= *REORG_THRESHOLD);
 
-        // Create Ethereum block ingestor
-        let block_ingestor = graph_datasource_ethereum::BlockIngestor::new(
-            store.clone(),
-            eth_adapter.clone(),
-            *ANCESTOR_COUNT,
-            ethereum_network_name.to_string(),
-            &logger_factory,
-            block_polling_interval,
-        )
-        .expect("failed to create Ethereum block ingestor");
+        // Create Ethereum block ingestors and spawn a thread to run each
+        eth_adapters
+            .iter()
+            .cloned()
+            .for_each(|(network_name, eth_adapter)| {
+                let block_ingestor = graph_datasource_ethereum::BlockIngestor::new(
+                    stores.get(network_name).expect("network with name").clone(),
+                    eth_adapter,
+                    *ANCESTOR_COUNT,
+                    network_name.to_string(),
+                    &logger_factory,
+                    block_polling_interval,
+                )
+                .expect("failed to create Ethereum block ingestor");
 
-        // Run the Ethereum block ingestor in the background
-        tokio::spawn(block_ingestor.into_polling_stream());
+                // Run the Ethereum block ingestor in the background
+                tokio::spawn(block_ingestor.into_polling_stream());
+            });
     }
 
-    // Prepare a block stream builder for subgraphs
-    let block_stream_builder = BlockStreamBuilder::new(
-        store.clone(),
-        store.clone(),
-        eth_adapter.clone(),
-        node_id.clone(),
-        *REORG_THRESHOLD,
-    );
+    let mut block_stream_builders = HashMap::new();
```

Comment:
I haven't checked the internals of the stream and host builders yet but could we keep these to one instance each, have them take the stores and eth adapters map and then take the network name as a build parameter?